models/balancer/polygon/balancer_v2_polygon_trades.sql

```diff
@@ -0,0 +1,138 @@
+{{ config(
+    schema = 'balancer_v2_polygon',
+    alias = 'trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook='{{ expose_spells(\'["polygon"]\',
+                                "project",
+                                "balancer_v2",
+                                \'["mendesfabio", "jacektrocinski"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2021-06-19' %}
+
+WITH swap_fees AS (
+    SELECT
+        swap.evt_block_number,
+        swap.evt_tx_hash,
+        swap.evt_index,
+        SUBSTRING(swap.`poolId`, 0, 42) AS contract_address,
+        swap.evt_block_time,
+        MAX(fees.block_time) AS max_fee_evt_block_time
+    FROM
+        {{ source ('balancer_v2_polygon', 'Vault_evt_Swap') }} swap
+        LEFT JOIN {{ ref('balancer_v2_polygon_pools_fees') }} fees
+            ON fees.contract_address = SUBSTRING(swap.`poolId`, 0, 42)
+            AND fees.block_time <= swap.evt_block_time
+    GROUP BY 1, 2, 3, 4, 5
+),
```

Comment:
Is it intended to get the fee at the moment of the swap? On Dune V1 I did it this way:

- https://github.com/balancer-labs/dune-spellbook/blob/main/deprecated-dune-v1-abstractions/ethereum/balancer/view_trades.sql#L149

Did you take a look into this?